### ISSUE TITLE
Allow creating certificate from public key without CSR

### ISSUE THREAD

**Issue by acul009** [rustls/rcgen#222]
Hi,

i'd love to use rcgen to generate my certificates, but I have the following use case which seems like it is not supported:

- client generates keypair
- client sends public key to the server along with some metadata about itself
- server decides what kind of certificate to create for the client
- server creates certificate with the given public key <--- This is where I fail.
- server sends the finished certificate back to the client for further use

Is that already possible somehow?

**Comment by djc** [comment]
Why don't you want to use a CSR as the vehicle for sending "public key .. along with some metadata about itself"?

**Comment by acul009** [comment]
The system I'm building will have agents register with a public key and a request to join. A user may then create and sign a certificate, letting the agent join the network.

The agent will be identified by a UUID - if I created the ID on the server I wouldn't have the problem of checking if the id is already in use, as that might be a good way to compromise the system.

But that's honestly not that bad. To be honest, I thought a CSR would include more information I'd have to check first. After some more research I realized I was wrong.

Thanks for pointing me in the right direction again and sorry for wasting your time ;)
